### ISSUE TITLE
Using readlink instead of greadlink on arm serials mac(Darwin)

### ISSUE THREAD

**Issue by trganda** [joernio/joern#3862]
**Describe the bug**

After installing joern with the guide from https://docs.joern.io/installation/
```
mkdir joern && cd joern # optional
curl -L "https://github.com/joernio/joern/releases/latest/download/joern-install.sh" -o joern-install.sh
chmod u+x joern-install.sh
./joern-install.sh --interactive
```
And specified add symbol link to `/usr/local/bin`.

Run the `joern` in the terminal, I got this error info:

```
/usr/local/bin/joern: line 4: greadlink: command not found
Unable to find ./bin/repl-bridge, have you created the distribution?
```

To fix this problem, I have tried installing `greadlink` with `brew install greadlink`, but the brew will install the `readline` and warn the `greadlink` does not exist and the error still happens.

<img width="864" alt="image" src="https://github.com/joernio/joern/assets/62204882/c880b237-0b62-47c6-a25d-fcaeb5defa0a">

So I have to modify the `joern` script to use `readlink`

<img width="981" alt="image" src="https://github.com/joernio/joern/assets/62204882/455d246c-d35f-4f40-a54b-5c58cf4f2335">

Since there were a lot of scripts (e.g. `c2cpg.sh`) of joern using the statement below,  this problem will influence other scripts as well.

```
if [ "$(uname -s)" = "Darwin" ]; then
    SCRIPT_ABS_PATH=$(greadlink -f "$0")
else
    SCRIPT_ABS_PATH=$(readlink -f "$0")
fi
```

**To Reproduce**
1. Just run `joern` command on the arm serial mac

**Expected behavior**
Open the joern interactive shell without error.

**Screenshots**
<img width="929" alt="image" src="https://github.com/joernio/joern/assets/62204882/b0bea5e4-f313-48b5-946e-5a0eca93c862">

**Desktop (please complete the following information):**
 - OS: m1 max mac with sonoma 14.1.1 (23B81)
 - Uname: Darwin loaclhost 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000 arm64
 - Joern Version:  v2.0.170
 - Java version: 21.0.1
 - Homebrew: 4.1.21

**Comment by xavierpinho** [comment]
It [seems](https://apple.stackexchange.com/questions/464136/which-macos-version-introduced-readlink-f) that `readlink -f` has become available on macOS as recently as macOS Monterey 12.3 (released March 2022). We may want to drop the usage of `greadlink` eventually, then.

@trganda As far as I know, the Homebrew package to install `greadlink` is `coreutils`.

**Comment by trganda** [comment]
> As far as I know, the Homebrew package to install greadlink is coreutils.

Thanks for the reply, it's work.
